### ISSUE TITLE
[Messenger] Consuming messages from doctrine transport fails (exception thrown)

### ISSUE THREAD

**Issue by DenisMir** [symfony/symfony#32738]
**Symfony version(s) affected**: 4.3.2

**Description**  
I am trying to use the messenger component with the doctrine transport asynchronously. 
I have activated the doctrine transport by `MESSENGER_TRANSPORT_DSN=doctrine://default` in the `.env.local`. 
In the messenger config the transport is configured as well: 
```
framework:
    messenger:
        transports:
            async: '%env(MESSENGER_TRANSPORT_DSN)%'
        routing:
            'App\Message\SomeNotification': async
```

When dispatching the message in the controller with `$this->dispatchMessage(new SomeNotification('some content'));` everything is fine. The table `messenger_messages` of the doctrine transport gets created automatically and the message is saved in the table. 

When trying to consume the message with `./bin/console messenger:consume async` I'm getting the following error: 

```
 [Symfony\Component\Debug\Exception\FatalThrowableError]                                                                                                                                                                                                               
  Argument 2 passed to Symfony\Component\Messenger\Worker::__construct() must implement interface Symfony\Component\Messenger\MessageBusInterface, string given, called in ..../vendor/symfony/messenger/  
  Command/ConsumeMessagesCommand.php on line 190         
```

So the `$routableBus` in the `Symfony\Component\Messenger\Command\ConsumeMessagesCommand` is empty which leads to the crash when instantiating the `Worker`.

**Comment by xabbuh** [comment]
Can you create a small example application that allows to reproduce this issue?

**Comment by linaori** [comment]
_A bit of extra information after talking on slack._

The Generated service definition for the `ConsumeMessagesCommand` is created with an empty argument, which results in a string.
```xml
    <service id="console.command.messenger_consume_messages" class="Symfony\Component\Messenger\Command\ConsumeMessagesCommand" public="false">
      <tag name="console.command" command="messenger:consume"/>
      <tag name="console.command" command="messenger:consume-messages"/>
      <tag name="monolog.logger" channel="messenger"/>
      <argument></argument>
      <argument type="service" id="messenger.receiver_locator"/>
      <argument type="service" id="monolog.logger.messenger" on-invalid="null"/>
      <argument type="collection">
        <argument>async</argument>
      </argument>
      <argument type="service" id="messenger.retry_strategy_locator"/>
      <argument type="service" id="debug.event_dispatcher"/>
      <call method="setCachePoolForRestartSignal">
        <argument type="service" id="cache.messenger.restart_workers_signal"/>
      </call>
      <call method="setName">
        <argument>messenger:consume</argument>
      </call>
      <call method="setAliases">
        <argument type="collection">
          <argument>messenger:consume-messages</argument>
        </argument>
      </call>
    </service>
```
https://github.com/symfony/symfony/blob/327fb95828352ec9db929150393eb377bb7e6a03/src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php#L188-L190 

If no `bus` option is given, `$this->routableBus` is an empty string and it will try to inject this.

I don't know enough of the component to actually tell why this is going wrong. I can't find any compiler passes modifying this argument either and this is what's in the repository:

https://github.com/symfony/symfony/blob/327fb95828352ec9db929150393eb377bb7e6a03/src/Symfony/Bundle/FrameworkBundle/Resources/config/console.xml#L84-L88

**Comment by xabbuh** [comment]
There is the `MessengerPass` which sets this argument in `registerReceivers()`. Can you debug if this is doing anything?

**Comment by linaori** [comment]
One final piece information I'd like to add, the PR that introduced this change (#31748) requires the same version of both the framework bundle and messenger component. It seems that this requirement isn't changed in `composer.json` and therefore allows installing a newer messenger component than the framework bundle, resulting in incomplete configuration. Previously the messenger bundle configured that. 

Please verify if the FrameworkBundle is at the same version as your Messenger component, and if not, see if updating it fixes this issue.

**Comment by DenisMir** [comment]
Ok this was the error. After updating `symfony/framework-bundle` to 4.3.2 everything works as expected.

**Comment by OskarStark** [comment]
@linaori can you send a PR then? Or shall I?

**Comment by OskarStark** [comment]
cc @chalasr

**Comment by linaori** [comment]
@OskarStark while for newer versions it would be nice, it will still allow upgrading to a broken version, do it won't actually solve this exact issue. My proposal is to either fix the constructor or the method call where it fails, to filter out this particular scenario, by filtering out an empty string and throwing an error (prefer constructor)

**Comment by chalasr** [comment]
Not sure it's really worth it, but here is a fix proposal: #32792
